mlir/test/Dialect/MIOpen/lowering_filter_tensor_ckyx_cnhw_knhw_f16.mlir

```diff
@@ -0,0 +1,70 @@
+// This tests checks the following aspects of lowering component:
+// * Filter is transformed correctly
+
+// RUN: mlir-opt -miopen-lowering %s | FileCheck %s
+
+func @miopen_conv2d_ckyx_cnhw_knhw(%filter : memref<8x128x3x3xf16>, %input : memref<8x128x32x32xf16>, %output : memref<128x128x30x30xf16>) {
+  miopen.conv2d(%filter, %input, %output) {
+    arch = "gfx906",
+    num_cu = 64,
+    filter_layout = ["c", "k", "y", "x"],
+    input_layout = ["ci", "ni", "hi", "wi"],
+    output_layout = ["ko", "no", "ho", "wo"],
+    dilations = [1, 1],
+    strides = [1, 1],
+    padding = [0, 0]
+  } : memref<8x128x3x3xf16>, memref<8x128x32x32xf16>, memref<128x128x30x30xf16>
+  return
+}
+// CHECK-LABEL: func @miopen_conv2d
+// CHECK-NEXT:  miopen.transform
+// CHECK:       gridwise_gemm_argument_position = 0
+// CHECK:       names = ["gemmK"]
+// CHECK:       source_names = ["c", "y", "x"]
+// CHECK:       names = ["gemmM"]
+// CHECK:       source_names = ["k"]
+// CHECK-NEXT:  miopen.transform
```

Comment:
This test is naively added.

Are those logic impacted particularly by data type? If they are not impacted particularly by data type, we should not duplicate the entire `lowering_filter_tensor_ckyx_cnhw_knhw` just for `fp16`. Instead, you should look at the source code and see the aspects that worth adding a fp16 test case and add it to original test case rather than create and duplicate a new one.

Reply:
Let me make it clear: It looks to me that all test cases in this file is really checking if `memref<*f16>` can be accepted by `-miopen-lowering` stage. If that's the case, what you can do is to create this one test case and only check this additional test case in the `lowering_top_level.mlir` instead of duplicating all the test cases in here.